cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ConfigMapUtils.java

```diff
@@ -67,10 +67,11 @@ public static Map<String, String> generateMetricsAndLogConfigMapData(Reconciliat
             data.put(supportsLogging.logging().configMapKey(), supportsLogging.logging().loggingConfiguration(reconciliation, metricsAndLogging.loggingCm()));
         }
 
-        if (model instanceof SupportsMetrics supportMetrics) {
-            String parseResult = supportMetrics.metrics().metricsJson(reconciliation, metricsAndLogging.metricsCm());
+        // this is only for JMX Prometheus Exporter, because Strimzi Metrics Reporter configuration is in the Kafka configuration file
+        if (model instanceof SupportsMetrics supportMetrics && supportMetrics.metrics() instanceof JmxPrometheusExporterModel) {
```

Comment:
Would `supportsMetrics` make more sense?

```suggestion
        if (model instanceof SupportsMetrics supportsMetrics && supportsMetrics.metrics() instanceof JmxPrometheusExporterModel) {
```

Reply:
Yes that makes rense. Refactored